src/v/cluster/controller.cc

```diff
@@ -384,4 +385,39 @@ ss::future<> controller::stop() {
     });
 }
 
+/**
+ * This function provides for writing the controller log immediately
+ * after it has been created, before anything else has been written
+ * to it, and before we have started communicating with peers.
+ */
+ss::future<> controller::cluster_creation_hook() {
+    if (!config::node().seed_servers().empty()) {
+        // We are not on the root node
+        co_return;
+    } else if (
+      _raft0->last_visible_index() > model::offset{}
+      || _raft0->config().brokers().size() > 1) {
```

Comment:
will `_raft0->config().brokers().size() > 1` this part of the conditional disable the creation hook later on if we add some ability to create raft with with an initial set of brokers rather than relying on explicit joining to increase group size?

Reply:
Yes -- as/when we do that, the explicit initialize step needs to handle the bootstrap user creation.  If we keep both the explicit and implicit initialization modes (which I think we will do in the near term), then this code sticks around but there will be a second site in the new explicit initialization code that handles user bootstrap.